File: PaintSpring/src/main/Paint/ShapeClass/IShape.java
package com.Paint.Paint.ShapeClass;

public interface IShape {

    String Name = "";
    String id = "";

    double xP = 0;
    double yP = 0;

    double x = 70;
    double y = 50;

    String border = "black";
    double borderWidth = 5;
    boolean borderScaleEnabled = false;

    String fill = "white";
    boolean fillEnabled = false;

    double rotation = 0;
    double[] points = {};

}

File: PaintAngular/src/app/ishape.ts
export interface IShape {
    // standard attrs

    name:string;
    id:string;

    xP:number;
    yP:number;

    x:number;
    y:number;

    border: string;
    borderWidth: number;
    borderScaleEnabled: boolean;

    fill: string;
    fillEnabled: boolean;

    rotation: number;

    // additional attributes
    points: number[];
}

// export interface IShape {
//     // standard attrs
    
//     x:number
//     y:number
//     xP:number
//     yP:number
//     id:string
//     name:string

// }